### ISSUE TITLE
AttributeError with getGlobalParameterIds() and getGlobalParameterValues functions

### ISSUE THREAD

**Issue by Niels86** [sys-bio/roadrunner#351]
I just started using libraodrunner and run into a number of problems. Whenever I use any of the functions relating to parameters I get the following Attribute error:

`AttributeError                            Traceback (most recent call last)
<ipython-input-46-c250ece50903> in <module>()
      4 import roadrunner.testing as test
      5 rtest = test.getRoadRunner('feedback.xml')
----> 6 test1 =rtest.getGlobalParameterIds()
      7 test2 =rtest.getGlobalParameterValues()
      8 print(test1)

/home/user/Applications/anaconda2/envs/default2.7/lib/python2.7/site-packages/roadrunner/roadrunner.pyc in <lambda>(self, name)
   2475     __setattr__ = lambda self, name, value: _swig_setattr(self, RoadRunner, name, value)
   2476     __swig_getmethods__ = {}
-> 2477     __getattr__ = lambda self, name: _swig_getattr(self, RoadRunner, name)
   2478 
   2479     def __init__(self, *args):

/home/user/Applications/anaconda2/envs/default2.7/lib/python2.7/site-packages/roadrunner/roadrunner.pyc in _swig_getattr(self, class_type, name)
     76 
     77 def _swig_getattr(self, class_type, name):
---> 78     return _swig_getattr_nondynamic(self, class_type, name, 0)
     79 
     80 

/home/user/Applications/anaconda2/envs/default2.7/lib/python2.7/site-packages/roadrunner/roadrunner.pyc in _swig_getattr_nondynamic(self, class_type, name, static)
     71         return method(self)
     72     if (not static):
---> 73         return object.__getattr__(self, name)
     74     else:
     75         raise AttributeError(name)

AttributeError: type object 'object' has no attribute '__getattr__'`

I get this error while using my own model in any sbml Version and Level combinations ranging from Version 1 to Level 1 to Version 3 Level 1. The same error is given when using the "'feedback.xml" model.
My roadrunner version and package version are:

`1.4.6; Compiler: gcc 4.4.7, C++ version: 1; JIT Compiler: LLVM-3.3; Date: Jun  2 2016, 22:52:47; LibSBML Version: `5.13.0``

Any suggestions on what might be the cause and how to circumvent this problem without having to go modifying the parameters in the xml file itself? Any help is appreciated. I loaded the models both with default configuration settings.

**Comment by matthiaskoenig** [comment]
Hi @Niels86,

the function you want to call is available on the `roadrunner.ExecutableModel`. This is returned when you load a model with roadrunner. In the case of the test function you have to first get the ÈxecutableModel`. 

```
import roadrunner.testing as test
rtest = test.getRoadRunner('feedback.xml')
# you have to get the executable roadrunner model (this is normally returned when you load a model with roadrunner, but
# not in the case of the test function)

rmodel = rtest.getModel()
test1 = rmodel.getGlobalParameterIds()
test2 =rmodel.getGlobalParameterValues()
print(test1)
```

```
['J0_VM1', 'J0_Keq1', 'J0_h', 'J4_V4', 'J4_KS4']
```
The best 
Matthias

**Comment by Niels86** [comment]
Thx Matthias, using getModel solved my problem. I though I already made the model executable when loading it. I had to use getModel() both for the test function and after loading my own model. It surprises me since as you mentioned if you load an SBML model it should already be executable right? It also suprisses me that Species can be accessed without using getModel() first while for parameters I had to first use the getModel() function.
